examples/plot_fmri_data_example.py

```diff
@@ -0,0 +1,273 @@
+# The `preprocess_haxby` fucntion make the preprocessing of the Haxby dataset,
+# it outputs the preprocessed activation maps for the two conditions
+# 'face' or 'house' (contained in `X`), the conditions (in `y`),
+# the session labels (in `groups`) and the mask (in `masker`).
+# You may choose a subject in [1, 2, 3, 4, 5, 6]. By default subject=2.
+data = preprocess_haxby(subject=2)
+X, y, groups, masker = data.X, data.y, data.groups, data.masker
+mask = masker.mask_img_.get_fdata().astype(bool)
+
+#############################################################################
+# Initializing FeatureAgglomeration object that performs the clustering
+# -------------------------------------------------------------------------
+# For fMRI data taking 500 clusters is generally a good default choice.
+
+n_clusters = 500
+# Deriving voxels connectivity.
+shape = mask.shape
+n_x, n_y, n_z = shape[0], shape[1], shape[2]
+connectivity = image.grid_to_graph(n_x=n_x, n_y=n_y, n_z=n_z, mask=mask)
+# Initializing FeatureAgglomeration object.
+ward = FeatureAgglomeration(n_clusters=n_clusters, connectivity=connectivity)
+
+#############################################################################
+# Making the inference with several algorithms
+# --------------------------------------------
+
+#############################################################################
+# First, we try to recover the discriminative partern by computing
+# p-values from SVR decoder weights and a parametric approximation
+# of the distribution of these weights.
+
+# We precomputed the regularization parameter by CV (C = 0.1) to reduce the
+# computation time of the example.
+beta_hat, scale = standardized_svr(X, y, Cs=[0.1])
+pval_std_svr, _, one_minus_pval_std_svr, _ = pval_from_scale(beta_hat, scale)
+
+#############################################################################
+# Now, we compute p-values thanks to permutation tests applied to
+# 1/the weights of the SVR decoder or 2/the weights of the Ridge decoder.
+
+# To derive the p-values from the SVR decoder, you may change the next line by
+# `SVR_permutation_test_inference = True`. It should takes around 15 minutes.
+
+SVR_permutation_test_inference = False
+if SVR_permutation_test_inference:
+    # We computed the regularization parameter by CV (C = 0.1)
+    pval_corr_svr_perm_test, one_minus_pval_corr_svr_perm_test = \
+        permutation_test_cv(X, y, n_permutations=50, C=0.1)
+
+# A faster solution is to compute the p-values by permutation test from the
+# Ridge decoder. The computation time is much shorter: around 20 seconds.
+
+estimator = Ridge()
+pval_corr_ridge_perm_test, one_minus_pval_corr_ridge_perm_test = \
+    permutation_test(X, y, estimator=estimator, n_permutations=200)
+
+#############################################################################
+# Now, let us run the Gaonkar algorithm (c.f. References).
+beta_hat, scale = gaonkar(X, y)
```

Comment:
Could we find a name different from `gaonkar` ? I think that this is a bit rude.